### PR TITLE
[Impeller] Support user defined structs in buffers, clean up compute tests

### PR DESCRIPTION
**Pull request by dnfield**
Updates the compute test to add a struct
Updates compiler reflector to support structs in buffers
Updates compute playground tests to not setup a renderer/attempt to setup a renderer.

### CODE REVIEW

File: impeller/renderer/compute_unittests.cc [hunk 1]
```diff
@@ -17,17 +17,10 @@
 namespace impeller {
 namespace testing {
 
-using ComputeTest = PlaygroundTest;
-INSTANTIATE_PLAYGROUND_SUITE(ComputeTest);
+using ComputeTest = ComputePlaygroundTest;
+INSTANTIATE_COMPUTE_SUITE(ComputeTest);
 
 TEST_P(ComputeTest, CanCreateComputePass) {
-  if (GetParam() == PlaygroundBackend::kOpenGLES) {
```
Comment (chinmaygarde) [inline impeller/renderer/compute_unittests.cc:24]: Did you mean to remove these skips?

**Review by chinmaygarde — APPROVED** [review]

File: impeller/compiler/reflector.cc [hunk 2]
```diff
@@ -603,6 +603,29 @@ std::vector<StructMember> Reflector::ReadStructMembers(
...
+          GetMemberNameAtIndex(struct_type, i),  // name
+          struct_member_offset,                  // offset
+          size,                                  // size
+          stride * array_elements.value_or(1),   // byte_length
+          array_elements,                        // array_elements
+          element_padding,                       // element_padding
+      });
+      current_byte_offset += stride * array_elements.value_or(1);
+      continue;
+    }
```
Comment (bdero) [inline impeller/compiler/reflector.cc:627]: I think this adds support for nested structs across all backend and stage types? Might be worth adding a small raster test that uses a nested struct (and make sure the GLES metadata doesn't need to be updated for this).

**Review by bdero — APPROVED** [review]
Yo dawg I heard you like structs.

File: impeller/renderer/compute_unittests.cc (hunk 1 above)
Comment (dnfield) [reply impeller/renderer/compute_unittests.cc:24]: Yes - this now gets skipped in the macro for setting up a compute playground. We can add Vulkan and GLES when/if we start supporting them.

File: impeller/compiler/reflector.cc (hunk 2 above)
Comment (dnfield) [reply impeller/compiler/reflector.cc:627]: Proooobably?

I'll give it a try.

Comment (dnfield) [reply impeller/compiler/reflector.cc:627]: I'm going to punt on this fornow - I'm still trying to track down what to do about the buffer to texture issue and want to keep making progress on compute right now.

**Pull request merged**